### PR TITLE
Fix repository and project name

### PR DESCRIPTION
**Pull request by dnicolodi**

### CODE REVIEW

File: tests/test_metadata.py [hunk 1]
```diff
@@ -32,8 +32,8 @@ def test_pep621(sdist_full_metadata):
         Classifier: Programming Language :: Python
         Project-URL: Homepage, https://example.com
         Project-URL: Documentation, https://readthedocs.org
-        Project-URL: Repository, https://github.com/mesonbuild/mesonpy
-        Project-URL: Changelog, https://github.com/mesonbuild/mesonpy/blob/master/CHANGELOG.md
+        Project-URL: Repository, https://github.com/mesonbuild/meson-python
+        Project-URL: Changelog, https://github.com/mesonbuild/meson-python/blob/master/CHANGELOG.md
```
Comment (rgommers) [inline tests/test_metadata.py:36]: this should be `.rst`

File: tests/packages/full-metadata/pyproject.toml [hunk 2]
```diff
@@ -37,8 +37,8 @@ test = [
 [project.urls]
 homepage = 'https://example.com'
 documentation = 'https://readthedocs.org'
-repository = 'https://github.com/mesonbuild/mesonpy'
-changelog = 'https://github.com/mesonbuild/mesonpy/blob/master/CHANGELOG.md'
+repository = 'https://github.com/mesonbuild/meson-python'
+changelog = 'https://github.com/mesonbuild/meson-python/blob/master/CHANGELOG.md'
```
Comment (rgommers) [inline tests/packages/full-metadata/pyproject.toml:41]: this should be `.rst`

File: mesonpy/_tags.py [hunk 3]
```diff
@@ -79,11 +79,7 @@ def __init__(self, value: str) -> None:
         self._additional_information = parts[2:]
 
         if self.implementation != 'cpython' and not self.implementation.startswith('pypy'):
-            raise NotImplementedError(
-                f'Unknown Python implementation: {self.implementation}. '
-                'Please report this to https://github.com/mesonbuild/mesonpy/issues '
-                'and include information about the Python distribution you are using.'
-            )
```
Comment (rgommers) [inline mesonpy/_tags.py:86]: This seems okay to me; this was useful at the start, but now it's just one of many low-probability things that someone should raise an issue for.

**Review by rgommers — APPROVED** [review]
Wow, there were a lot of left-overs. Thanks for cleaning that up @dnicolodi. All LGTM module one small link issue.

File: mesonpy/_tags.py (hunk 3 above)
Comment (dnicolodi) [reply mesonpy/_tags.py:86]: Do you mean okay to leave as it is, or okay to simplify?

Comment (rgommers) [reply mesonpy/_tags.py:86]: leave as is. I'll merge this now, CI was happy already

**Comment by rgommers** [comment]
Actually, remembering we want a clean linear history, can you please do two things @dnicolodi?

1. Rewrite the history so the `MAINT: fix references to CHANGELOG.rst` fixup commit is merged with `MAINT: fix repository name`
2. Tweak the commit message of the third commit: `BUILD` -> `BLD`

**Comment by dnicolodi** [comment]
Sure. I kept the commit separate because the issue seemed unrelated, but I'm happy to merge them.

**Comment by dnicolodi** [comment]
I think something is off with the read-the-docs setup. The CI job never reports back. I think it may be related to the change of git repository. It is a requirement for the auto-merge thus this PR will not be merged automatically.

**Comment by rgommers** [comment]
> I think something is off with the read-the-docs setup. The CI job never reports back. I think it may be related to the change of git repository. It is a requirement for the auto-merge thus this PR will not be merged automatically.

Not sure what is going on there, I've now manually rebase-merged your PR, thanks again!

@FFY00 would you be able to have a look at ReadTheDocs? It probably needs a simple tweak, looks like it has been failing since the repo move to the mesonbuild org. Also, my RTD username is `rgommers` in case you want to add me there.

**Pull request merged**